packages/@tinacms/graphql/src/resolver/index.ts

```diff
@@ -835,13 +838,34 @@ export class Resolver {
               // load the doc with the references
               let refDoc = await this.getRaw(pathToDocWithRef)
 
+              let hasUpdate = false
               // Update each reference to the deleted document
               for (const path of referencePaths) {
-                refDoc = updateObjectWithJsonPath(refDoc, path, realPath, null)
+                const { object, updated } = updateObjectWithJsonPath(
+                  refDoc,
+                  path,
+                  realPath,
+                  null
+                )
+                refDoc = object
+                hasUpdate = updated || hasUpdate
               }
 
-              // save the updated doc
-              await this.database.put(pathToDocWithRef, refDoc, collection)
```

Comment:
this was causing the issue (adding the doc with references to the referenced collection)